akka-distributed-data/src/main/scala/akka/cluster/ddata/VersionVector.scala

```diff
@@ -184,10 +248,19 @@ final case class VersionVector private[akka] (
     }
 
     if ((this eq that) || (this.versions eq that.versions)) Same
-    else compare(this.versions.iterator, that.versions.iterator, if (order eq Concurrent) FullOrder else order)
+    else compare(this.versionsIterator, that.versionsIterator, if (order eq Concurrent) FullOrder else order)
   }
 
   /**
+   * INTERNAL API
+   */
+  private[akka] def versionsIterator: Iterator[(UniqueAddress, Long)] =
+    versions match {
+      case Right(tuple)  ⇒ Iterator.single(tuple)
```

Comment:
@viktorklang Do you mean like this: https://github.com/akka/akka/commit/2a65671070a6c1672d53ee8f449b0d50a4873948